tools/xibuild/Main.cs

```diff
@@ -0,0 +1,220 @@
+
+			Console.WriteLine ($"Running xibuild with args: {String.Join (" ", args)}");
+			List<string> remaining = null;
+
+			try {
+				remaining = p.Parse (args);
+			} catch (OptionException oe) {
+				Console.WriteLine (oe.Message);
+				return -1;
+			}
+
+			if (configGenerationOnly && runTool) {
+				Console.WriteLine ("Use either -c or -t, but not both.\n");
+				ShowHelp ();
+				return -1;
+			}
+
+			if (!String.IsNullOrEmpty (baseConfigFile) && !File.Exists (baseConfigFile)) {
+				Console.WriteLine ($"Base config file {baseConfigFile} not found.");
+				return -1;
+			}
+
+			if (configGenerationOnly && remaining.Count == 0) {
+				Console.WriteLine ("Please specify the path to managed tool to generate an app.config file for it.");
+				return -1;
+			}
+
+			if (runTool && remaining.Count == 0) {
+				Console.WriteLine ("Please specify the path to managed tool to run.");
+				return -1;
+			}
+
+			bool runMSBuild = !runTool && !configGenerationOnly;
+
+			if (!runTool && !runMSBuild) {
+				GenerateAppConfig (remaining [0] + ".config", baseConfigFile);
+				return 0;
+			}
+
+			return RunTool (
+					toolPath: runMSBuild ? "msbuild" : remaining [0],
+					combinedArgs: BuildQuotedCommandLine (remaining, runMSBuild ? 0 : 1),
+					baseConfigFile: runMSBuild ? null : baseConfigFile);
+
+			string BuildQuotedCommandLine (List<string> a, int skip) => String.Join (" ", a.Skip (skip).Select (arg => $"\"{arg}\""));
+
+			void ShowHelp ()
+			{
+				Console.WriteLine ("xibuild: Run msbuild or a tool with a custom msbuild config file which adds fallback paths from MSBuildExtensionsPathFallbackPathsOverride.\n");
+				Console.WriteLine ("Usage: xibuild [-c] [-t] [-m <base config file>] [-h] -- [path to managed tool] [arguments]");
+				Console.WriteLine ("Default: Generate a temporary app.config file and run msbuild");
+				Console.WriteLine ();
+				Console.WriteLine ("\t-c: Generate config file only");
+				Console.WriteLine ("\t-t: Path to the managed tool to run. If this and `-c` are not used, then this runs msbuild");
+				Console.WriteLine ("\t-m <base config file>: Config file to merge with the one from msbuild.dll.config");
+				Console.WriteLine ("\t-h: help");
+				Console.WriteLine ("\nNote: Adds the path from the environment variable MSBuildExtensionsPathFallbackPathsOverride to the list of fallback paths in the generated app.config");
+				Console.WriteLine ("\nExamples:");
+				Console.WriteLine ("xibuild -c /path/to/foo.exe : Generate /path/to/foo.exe.config");
+				Console.WriteLine ("xibuild -- /v:diag /path/to/project.csproj : Run msbuild with a custom app.config and the arguments passed to msbuild.");
+				Console.WriteLine ("xibuild -t -- /path/to/managed_tool.exe args : Run managed_tool.exe with the arguments, and a custom app.config");
+				Console.WriteLine ("");
+				Console.WriteLine ("Add `-m /path/to/base.exe.config` to merge the generated app.config with base.exe.config .");
```

Comment:
Minor comment, but I think you could have used the options print method, so that the help and the options are in sync. Something like

```csharp
options.WriteOptionDescriptions (Console.Out);
```